system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs

```diff
@@ -372,15 +379,24 @@ pub type NativeAndAssets = fungible::UnionOf<
 	AccountId,
 >;
 
+pub type PoolIdToAccountId = pallet_asset_conversion::AccountIdConverter<
+	AssetConversionPalletId,
+	(xcm::v3::Location, xcm::v3::Location),
+>;
+
 impl pallet_asset_conversion::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type Balance = Balance;
 	type HigherPrecisionBalance = sp_core::U256;
 	type AssetKind = xcm::v3::Location;
 	type Assets = NativeAndAssets;
 	type PoolId = (Self::AssetKind, Self::AssetKind);
-	type PoolLocator =
-		pallet_asset_conversion::WithFirstAsset<KsmLocationV3, AccountId, Self::AssetKind>;
+	type PoolLocator = pallet_asset_conversion::WithFirstAsset<
+		KsmLocationV3,
+		AccountId,
+		Self::AssetKind,
+		PoolIdToAccountId,
+	>; // FAIL-CI @muharem does this make sense or do we need the AccountIdConverter?
```

Comment:
@muharem i copied this from the SDK, please check.

Reply:
for Kusama we need to leave it with no seed for now (`AccountIdConverterNoSeed`) to keep the same derivation as now. to migrate to new derivation, we need to include the migration pallet. I can do it in separate PR, this is not critical.

Reply:
in Polkadot Asset Hub, for the same type parameter we can use this setup, and remove the `imps::pool` module. this is because for the pallet for Polkadot was launched with the new derivation.

Reply:
Not sure if i understood this correctly, but i could not get it to compile with the changes.    
Can you please just push a commit here to fix it?